src/nodes.coffee

```diff
@@ -1431,9 +1430,8 @@ exports.Param = class Param extends Base
     return @reference if @reference
     node = @name
     if node.this
-      node = node.properties[0].name
-      if node.value.reserved
-        node = new Literal o.scope.freeVariable node.value
+      name = "at_#{node.properties[0].name.value}"
```

Comment:
Ok, the "issue" is that "_slice" is getting validated and passes the validation. Then, after the validation, `Param::asReference` changes "_slice" into "__slice". At that point no more validation is done, so it slips through. If the __slice helper hasn't been used yet there are no scope checks to stop it either. So the bug (if any) is that `Scope::freeVariable` does iterate until the var name is also not a reserved word.